autogluon_zeroshot/contexts/utils.py

```diff
@@ -1,26 +1,32 @@
 from . import intersect_folds_and_datasets
+from ..simulation.tabular_predictions import TabularPicklePredictions, TabularPicklePredictionsOpt
 
 
-# TODO: Prune zeroshot_gt based on zeroshot_pred_proba final datasets
 def load_zeroshot_input(path_pred_proba, path_gt, zsc, lazy_format: bool = False):
+    print(f'Loading ZS inputs:\n'
+          f'\tpred_proba:  {path_pred_proba}\n'
+          f'\tgt:          {path_gt}\n'
+          f'\tlazy_format: {lazy_format}')
     zeroshot_gt = zsc.load_groundtruth(path_gt=path_gt)
     zeroshot_pred_proba = zsc.load_pred(
         pred_pkl_path=path_pred_proba,
         lazy_format=lazy_format,
     )
-    if lazy_format:
-        raise NotImplementedError('lazy_format=True not yet supported for this function.'
-                                  'Need to add `force_to_dense` method')
 
     # keep only dataset whose folds are all present
     intersect_folds_and_datasets(zsc, zeroshot_pred_proba, zeroshot_gt)
-    zeroshot_pred_proba.force_to_dense(prune_method='dataset')
+    zeroshot_pred_proba.force_to_dense(first_prune_method='task', second_prune_method='dataset')
 
     zsc.subset_models(zeroshot_pred_proba.models)
     zsc.subset_datasets(zeroshot_pred_proba.datasets)
     zeroshot_pred_proba.restrict_models(zsc.get_configs())
     zeroshot_gt = prune_zeroshot_gt(zeroshot_pred_proba=zeroshot_pred_proba, zeroshot_gt=zeroshot_gt)
 
+    # Convert to optimized format
+    if isinstance(zeroshot_pred_proba, TabularPicklePredictions) and \
```

Comment:
Do we need the type test? I did not see how the type could be `TabularPicklePredictionsOpt` at this point.

Reply:
Technical not, but it is possible `zsc.load_pred` could in future return `TabularPicklePredictionsOpt`. At minimum the type check doesn't hurt.